src/pages/api/delete-identity.ts

```diff
@@ -15,42 +15,54 @@ import { logger } from "@src/logger";
 import { withAuth } from "@src/middlewares/with-auth";
 import { withSentry } from "@src/middlewares/with-sentry";
 import getTracer from "@src/tracer";
-
-const tracer = getTracer();
+import { ERRORS_DATA, webErrorFactory } from "@src/web-errors";
 
 const handler: Handler = (request, response): Promise<void> => {
-  return tracer.span("delete-identity handler", async (span) => {
+  const webErrors = {
+    badRequest: ERRORS_DATA.BAD_REQUEST,
+    unexpectedError: ERRORS_DATA.UNEXPECTED_ERROR,
+  };
+
+  return getTracer().span("delete-identity handler", async (span) => {
     const { userId, type, value } = request.body;
 
+    if ([userId, type, value].includes(undefined)) {
+      throw webErrorFactory(webErrors.badRequest);
+    }
+
     span.setDisclosedAttribute("Identity type", type);
 
     return removeIdentityFromUser(config.managementCredentials, {
       userId,
       identityType: type,
       identityValue: value,
-    }).then(() => {
-      span.setDisclosedAttribute("is Identity removed", true);
-
-      response.statusCode = HttpStatus.ACCEPTED;
-      response.setHeader("Content-Type", "application/json");
-      response.end();
-      return;
-    });
+    })
+      .then(() => {
+        span.setDisclosedAttribute("is Identity removed", true);
+
+        response.statusCode = HttpStatus.ACCEPTED;
+        response.setHeader("Content-Type", "application/json");
+        response.end();
+        return;
+      })
+      .catch(() => {
+        throw webErrorFactory(webErrors.unexpectedError);
```

Comment:
this could certainly be separated into several errors, no?
`connect_unreachable`, `identity_not_found` or things like that

Reply:
I'm not sure how to separate them as all the `removeIdentityFromUser` function throws are 
```
  if (errors) {
    throw new GraphqlErrors(errors);
  }
 ```
 
 🤔

Reply:
```
> fetch("http://undefineddomaine.com")
Promise { <pending> }
> (node:42415) UnhandledPromiseRejectionWarning: FetchError: request to http://undefineddomaine.com/ failed, reason: getaddrinfo ENOTFOUND undefineddomaine.com
```

You could deal with `FetchError` and `GraphqlErrors` for now.

Reply:
Would be great to fine tune the error game of `connect-js-management` regarding `GraphQLErrors`, but not sure it is worth for now as the errors of `management` are not documented.

Reply:
Looks like I don't have access to `FetchError ` 🤔.

Reply:
`FetchError` should be importable from `node-fetch` or `cross-fetch` no?

> Would be great to fine tune the error game of connect-js-management regarding GraphQLErrors, but not sure it is worth for now as the errors of management are not documented.

I agree we don't need everything now

Reply:
Just a follow up, `cross-fetch` do not export a `FetchError` object :/.

Reply:
Alright, so `FetchError` can be found in `@types/node-fetch`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-fetch/index.d.ts

I can have access to the type if I install `@types/node-fetch` in `connect-js-management`, `node-fetch` being used by `cross-fetch`.

Reply:
I found the PR regarding the addition of `cross-fetch`: https://github.com/fewlinesco/connect-account/pull/129/files.
We can use `node-fetch` instead on `connect-js/management` is you want.

Reply:
Nvm, I tried it, and I remember why we use `cross-fetch` instead of `node-fetch`:

`fetchManagement.ts`
<img width="733" alt="image" src="https://user-images.githubusercontent.com/31956107/107775922-39812980-6d41-11eb-8134-6064c455b7f5.png">